PurchasesTests/SubscriberAttributes/BackendSubscriberAttributesTests.swift

```diff
@@ -297,4 +297,48 @@ class BackendSubscriberAttributesTests: XCTestCase {
 
         expect(requestBody["attributes"]).to(beNil())
     }
+
+    func testPostReceiptWithSubscriberAttributesPassesErrorsToCallback() {
+        var completionCallCount = 0
+
+        self.mockHTTPClient.stubbedCompletionStatusCode = 400
+        let attributeErrors = [
+            RCAttributeErrorsKey: ["$email": "email is not in valid format"]
+        ]
+        let attributesErrorsResponse = [
+            RCAttributeErrorsResponseKey: attributeErrors
+        ]
+        self.mockHTTPClient.stubbedCompletionResponse = attributesErrorsResponse
+
+        let subscriberAttributesByKey: [String: RCSubscriberAttribute] = [
+            subscriberAttribute1.key: subscriberAttribute1,
+            subscriberAttribute2.key: subscriberAttribute2
+        ]
+        var receivedError: NSError? = nil
+        backend.postReceiptData(receiptData,
+                                appUserID: appUserID,
+                                isRestore: false,
+                                productIdentifier: nil,
+                                price: nil,
+                                paymentMode: .none,
+                                introductoryPrice: nil,
+                                currencyCode: nil,
+                                subscriptionGroup: nil,
+                                discounts: nil,
+                                presentedOfferingIdentifier: nil,
+                                observerMode: false,
+                                subscriberAttributes: subscriberAttributesByKey,
+                                completion: { (purchaserInfo, error) in
+                                    completionCallCount += 1
+                                    receivedError = error as NSError?
+                                })
+
+        expect(self.mockHTTPClient.invokedPerformRequestCount) == 1
+
+        expect(receivedError).toNot(beNil())
+        guard let nonNilReceivedError = receivedError else { fatalError() }
+        expect(nonNilReceivedError.successfullySynced()) == true
```

Comment:
@vegaro in this test case I'm intentionally not setting the finishable key to make sure that this works correctly when it's not present, and that successfullySyncedKey is set in userInfo